controllers/user-routes.js

```diff
@@ -52,19 +52,41 @@ router.get('/newPost', async (req, res) => {
         if (user) {
            res
             .status(200)
-            .render('new-post', { user }); 
+            .render('add-post') 
         }
 
-        if(!user){
+        else if(!user){
             res
-            .status(400)
-            .render('new-post', { message: 'Please login or register to make a new post.'});
+            .status(200)
+            .render('add-post', { message: 'Please login or register to make a new post.'})
         }
     } catch (err) {
       console.log(err);
         res.status(500).json(err);
     }
   });
 
+//post route for making a new post inserting into db
+router.post('/newPost', async (req, res) => {
+  const user = req.session
+  try {
+    if (user){
+      res
+        .status(400)
+        .redirect('/login', {message: 'please login to make a new post'})
+    }
+    if (!user){
+      const dbPostData = await Post.create({
+        title: req.body.title,
+        entry: req.body.entry
+    })
+      res
+        .status(200)
+        .json(dbPostData)
+    }
+  } catch (err) {
+    console.log(err)
+  }
+})
```

Comment:
I don't think this logic does what you think,
Line 73: If there is a "user session" please login?
Line 78: If there is a NOT a "user session" post an entry?
Seems backwards to me.